include/libcyphal/executor.hpp

```diff
@@ -204,6 +89,27 @@ class IExecutor : public cetl::rtti_helper<IExecutorTypeIdType>
         ///
         using Function = cetl::pmr::function<void(const TimePoint now_time), FunctionMaxSize>;
 
+        /// @brief Defines maximum size of callback implementation.
+        ///
+        /// Size is chosen arbitrary, but it should be enough to store any callback implementation.
+        ///
+        static constexpr std::size_t MaxSize = (sizeof(void*) * 10) + sizeof(Function);
+
+        /// @brief Defines type-erased callback which is capable to store some internal implementation.
+        ///
+        /// It is supposed to be used to hide callback implementation,
+        /// and to be passed to the executor for scheduling.
+        /// It also manages lifetime and ownership of the callback registration, so it's movable but not copyable.
+        ///
+        using Any = cetl::unbounded_variant<MaxSize, false /* Copyable */, true /* Movable */>;
+
+        /// @brief Defines opaque handle to a registered callback.
+        ///
+        /// It is supposed to be used internally (by derived executors)
+        /// to identify/reference the callback in the executor.
+        ///
+        using Handle = std::uintptr_t;
```

Comment:
Casting pointer-to-int and back looks sus. Obviously there's no casting going on here but the `uintptr` named "Handle" obviously invites one to do that. Why do we need this in the interface considering that it's not used by the interface's clients nor the interface itself? If an implementation needs to do `reinterpret_cast<>` (which is not good in itself but that is beside the point), let it define its own Handle type.

Keeping it here contaminates the interface with implementation details, I suggest removing it.